ecgtools/core.py

```diff
@@ -259,8 +261,15 @@ def build(
             'aggregations': aggregations,
         }
 
-        filelist = self.filelist or self._get_filelist_from_dirs()
-        df = parse_files_attributes(filelist, self.parser, self.lazy, self.nbatches)
+        if len(self.filelist) == 0:
+            filelist = self.filelist or self._get_filelist_from_dirs()
+        else:
+            filelist = self.filelist
```

Comment:
How is this block different from this single line below?

```python
filelist = self.filelist or self._get_filelist_from_dirs()
```

Reply:
@kmpaul thanks for pointing this out as it was an oversight on my part.  I will make this change.